app/controllers/charge_controller.js

```diff
@@ -45,16 +46,17 @@ const appendChargeForNewView = async function appendChargeForNewView (charge, re
     prepaidCredit: charge.gatewayAccount.corporatePrepaidCreditCardSurchargeAmount,
     prepaidDebit: charge.gatewayAccount.corporatePrepaidDebitCardSurchargeAmount
   })
-  charge.post_card_action = routeFor('create', chargeId)
   charge.id = chargeId
+  charge.exampleCardExpiryDateYear = getTwoDigitYearInFuture()
```

Comment:
GitHub discarded my previous comment and I've lost the will so less details now 🤦 .

We're conflating the view context and the charge (already done before this PR, this file is some of the most confusing code I've ever read!), assigning this payment page specific value to something called `charge` is confusing. 

This information could be passed to the view context at the point of rendering, unfortunately that happens 3 times in this controller because of how it's organised. 

It would look something like:
```js
responseRouter.response(req, res, views.CHARGE_VIEW, withAnalytics(charge, { charge, exampleCardExpiryDateYear })
```

Also `withAnalytics(charge, charge)` 🤨 

As I mentioned this PR doesn't cause this behaviour so it shouldn't be required to address this but the data definitely doesn't belong to the charge.

Reply:
Oh yes, I felt terrible adding `exampleCardExpiryDateYear` to the `charge` but did it because that was the existing pattern. However, while most of the properties `charge` is burdened with can vary between payments (and therefore be considered to belong to the `charge` if you squint a bit), there’s nothing at all payment-specific about the example expiry date. I’ll have a look at doing what you suggest.

Reply:
I did start doing this but it seemed to have way more ripple effects than I anticipated, so I think we’ll have to leave it for now.